File: 1.java
public class ExampleSonarIssues {

    public static void main(String[] args) {
        ExampleSonarIssues obj = new ExampleSonarIssues();
        obj.unusedMethod();
        obj.methodWithTooManyParams(1, 2, 3, 4, 5, 6);
        obj.potentialNullPointerRisk();
    }

    // Issue 1: Unused method
    public void unusedMethod() {
        System.out.println("This method is never called.");
    }

    // Issue 2: Method with too many parameters (bad design)
    public void methodWithTooManyParams(int a, int b, int c, int d, int e, int f) {
        System.out.println("Too many parameters make this hard to maintain.");
    }

    // Issue 3: Hardcoded credentials (security risk)
    public void hardcodedPassword() {
        String password = "12345"; // Sensitive data hardcoded
        System.out.println("The password is: " + password);
    }

    // Issue 4: Potential NullPointerException
    public void potentialNullPointerRisk() {
        String riskyString = null;
        if (riskyString.length() > 5) { // This will throw a NullPointerException
            System.out.println("Risky string length: " + riskyString.length());
        }
    }

    // Issue 5: Commented-out code
    public void commentedCodeExample() {
        // System.out.println("This code is commented out and should be removed.");
    }

    // Issue 6: Inefficient string concatenation in a loop
    public void inefficientStringConcatenation() {
        String result = "";
        for (int i = 0; i < 10; i++) {
            result += i; // Inefficient use of string concatenation in a loop
        }
        System.out.println(result);
    }
}

File: 1.py
class ExampleSonarIssues:
    def __init__(self):
        self.password = "12345"  # Hardcoded sensitive information (security risk)

    def unused_function(self):
        # This function is never called, representing dead code
        print("This is an unused function.")

    def function_with_many_parameters(self, a, b, c, d, e, f):
        # Too many parameters make the code hard to maintain
        print(f"Received: {a}, {b}, {c}, {d}, {e}, {f}")

    def potential_divide_by_zero(self, divisor):
        # Possible division by zero without validation
        result = 100 / divisor
        print(f"Result i      s {result}")

    def inefficient_loop(self):
        # Inefficient string concatenation in a loop
        result = ""
        for i in range(10):
            result += str(i)
        print(result)

    def commented_out_code(self):
        # print("This code is commented out and should be removed.")
        pass

    def risky_variable(self):
        # Using a variable before checking if it's None
        risky_string = None
        if len(risky_string) > 5:  # This will throw a TypeError
            print("Risky string is long enough.")


# Instantiate the class and call methods to simulate code execution
example = ExampleSonarIssues()
example.function_with_many_parameters(1, 2, 3, 4, 5, 6)
example.potential_divide_by_zero(0)  # Deliberate divide-by-zero error
example.inefficient_loop()
example.risky_variable()
